sagemaker-spark-sdk/src/main/scala/com/amazonaws/services/sagemaker/sparksdk/protobuf/ProtobufConverter.scala

```diff
@@ -194,4 +212,67 @@ object ProtobufConverter {
     protobufBuilder.addFeatures(mapEntry)
   }
 
+  /*
+   *  DenseMatrix is stored in column major as
+   *      Record.values([value_0, value_1, value_2, ...])
+   *      Record.shape([rows, cols])
+   *
+   *  SparseMatrix is stored as Compressed Sparse Column (CSC)
+   *  (See Spark SparseMatrix documentation)
+   *      Record.values([value_0, value_1, value_2, ...])
+   *      Record.shape([rows, cols])
+   *      Record.keys([rowIndices_size, colPtrs_size, rowIndicies, colPtrs])
+   *
+   *  For instance the SparseMatrix[ 1.0 0.0 4.0
+   *                                 0.0 3.0 5.0
+   *                                 2.0 0.0 6.0 ] is encoded as:
+   *     Record.values[1.0, 2.0, 3.0, 4.0, 5.0, 6.0]
+   *     Record.shape[3, 3]
+   *     Record.keys[6, 4, 0, 2, 1, 0, 1, 2, 0, 2, 3, 6]
+   */
+  private def setFeatures(protobufBuilder: Record.Builder,
+                          matrix: Matrix): Record.Builder = {
+    if (matrix.isTransposed) {
+      throw new IllegalArgumentException(s"Transposed Matrices are not supported")
+    }
+
+    val featuresTensorBuilder = Value.newBuilder().getFloat32TensorBuilder()
+
+    // Matrix shape must be recorded for both dense/sparse matrices
+    featuresTensorBuilder.addShape(matrix.numRows)
+    featuresTensorBuilder.addShape(matrix.numCols)
+
+    val featuresTensor = matrix match {
+      case m: DenseMatrix =>
+        for (value <- m.values) {
+          featuresTensorBuilder.addValues(value.toFloat)
+        }
+
+        featuresTensorBuilder.build()
+
+      case m: SparseMatrix =>
+        for (value <- m.values) {
+          featuresTensorBuilder.addValues(value.toFloat)
+        }
+
+        // Specify wheter the Sparse matrix is transposed
```

Comment:
Can you remove this comment? (It's never transposed)

Reply:
Absolutely.